src/com/dotmarketing/portlets/rules/business/RulesEngine.java

```diff
@@ -44,13 +50,32 @@ public static void fireRules(HttpServletRequest req, HttpServletResponse res, Ru
 
         try {
 
-            Set<Rule> rules = APILocator.getRulesAPI().getRulesByParentFireOn(host.getIdentifier(), systemUser, false, fireOn);
-
+            Set<Rule> rules = APILocator.getRulesAPI().getRulesByParentFireOn(parent.getIdentifier(), systemUser, false, fireOn);
+			List<Rule> firedRules = (List<Rule>) req.getAttribute(WebKeys.RULES_ENGINE_FIRE_LIST);
+        	if(firedRules ==null){
+        		firedRules = new ArrayList<Rule>();
+        		req.setAttribute(WebKeys.RULES_ENGINE_FIRE_LIST, firedRules);
+        	}
```

Comment:
right, I thought that too.  I just am wary of adding more to the session.  I was thinking we could use a flash session (which I have always thought we needed).  Maybe phase 2?